### ISSUE TITLE
Error wrapping node's `fs` in an OverlayFS due to nonstandard stat.clone() method

### ISSUE THREAD

**Issue by wmhilton** [jvilk/BrowserFS#208]
Welcome to the latest edition of William's extreme BrowserFS workout!

Here's what I expected to work:
<details>
<summary>mountFixture.js</summary>

```js
const _fs = require('fs')
const BrowserFS = require('browserfs')

const mountFixture = function (dir) {
  return new Promise((resolve, reject) => {
    BrowserFS.configure(
      {
        fs: 'OverlayFS',
        options: {
          readable: {
            fs: 'FolderAdapter',
            options: {
              folder: dir,
              wrapped: _fs
            }
          },
          writable: {
            fs: 'InMemory',
            options: {}
          }
        }
      },
      function (err) {
        if (err) {
          reject(err)
        } else {
          resolve(BrowserFS.BFSRequire('fs'))
        }
      }
    )
  })
}

module.exports.mountFixture = mountFixture
```

</details>

<details>
<summary>test.js</summary>

```js
const { mountFixture } = require('./mountFixture.js')
mountFixture(process.cwd())
.then(fs => {
  let files = fs.readdirSync('/')
  console.log(files)
})
```

</details>

It works if I just use `FolderAdapter` but when I try to use `OverlayFS` I get

```
TypeError: this._readable.statSync(...).clone is not a function
```

Testing a fix at the moment. Planned strategy is to use a static method on the class to do the cloning rather than assume the stat instance was made by us and has a non-standard `.clone`  method.

**Comment by jvilk** [comment]
Interesting; I hadn't anticipated people using BrowserFS like this in Node. A static method, `Stats.clone`, sounds like a good compromise. I cannot guarantee that this use case will work in all cases, though; a NodeWrapper backend might be a cleaner general-purpose solution.

**Comment by jvilk** [comment]
I added a fix. This will be in today's beta release. Hope it helps!

**Comment by wmhilton** [comment]
I'll check it out! I'd run into some other issue though once I got passed the instance clone method, so I switched tactics and am using temporary directories and copying all the files at the moment.

But I'd really like to use the OverlayFS(FolderAdaptor + MemoryFS) option in Node, because then I could achieve the same result without having to actually copy files around the disk. And then I can start using the OverlayFS(FolderAdaptor + MemoryFS) for my other project that involves doing clever things with "virtual" node_modules folders.
